src/lsptoolshost/copilot.ts

```diff
@@ -35,58 +38,60 @@ export async function registerCopilotExtensionAsync(
         }
         return;
     }
-    await ext.activate();
-    const relatedAPI = ext.exports as CopilotRelatedFilesProviderRegistration | undefined;
-    if (!relatedAPI) {
-        if (isTraceLogLevel) {
-            channel.appendLine(
-                'Incompatible GitHub Copilot extension installed. Skip registeration of C# related files provider.'
-            );
+    ext.activate().then(() => {
+        const relatedAPI = ext.exports as CopilotRelatedFilesProviderRegistration | undefined;
+        if (!relatedAPI) {
```

Comment:
Seems like this should work, but I'm not super confident on how it works in JS.  Can you verify that nothing throws if you have an older version of copilot with the previous version of the API?

Reply:
Hmm, it seems to work with old API as well,. My TS knowledge is poor so this is unexpected to me, I thought `relatedAPI ` for copilot extension with old API would be undefined. What I saw under debugger is the registration just work and callback when called got an undefined cancellation token.